src/core/frames/frame_controller.ts

```diff
@@ -160,7 +161,14 @@ export class FrameController
     try {
       const html = await fetchResponse.responseHTML
       if (html) {
-        const { body } = parseHTMLDocument(html)
+        let body
+        if (CSPTrustedTypesPolicy == null) {
+          body = parseHTMLDocument(html)
+        } else {
+          const trustedHTML = CSPTrustedTypesPolicy.createHTML(html)
+          body = parseHTMLDocument(trustedHTML as string)
+        }
```

Comment:
If we don't move the trustedTypes to `parseHTMLDocument`, we need to make sure `body` is correct

```suggestion
        if (CSPTrustedTypesPolicy == null) {
          { body } = parseHTMLDocument(html)
        } else {
          const trustedHTML = CSPTrustedTypesPolicy.createHTML(html)
          { body } = parseHTMLDocument(trustedHTML as string)
        }
```

Reply:
yes, we can do this